libmetal/lib/processor/hosted/cpu.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * @file	cpu.h
+ * @brief	Hosted test environment CPU specific primitives
+ */
+
+#include <metal/sleep.h>
+
+#ifndef __METAL_HOSTED_CPU__H__
+#define __METAL_HOSTED_CPU__H__
+
+static inline void metal_cpu_yield(void)
+{
+	__metal_cpu_yield();
```

Comment:
The metal cpu yield depend on the processor not the system.
iw ould be beter to create a ` metal_wait_used()` instead